### PR TITLE
User routes

### PR DESCRIPTION
**Pull request by podmar**
Hi Lina! Here are the register, login, updateProfile and uploadPhoto routes. Cheers Marta

### CODE REVIEW

File: server/tsconfig.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 {
   "compilerOptions": {
+    "typeRoots": ["./src/@types"],
```
Comment (linablum) [inline server/tsconfig.json]: Approach to access enums, redundant

File: server/src/@types/User.d.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    interface UserLoginData {
+        password: string;
+        email: string;
+    }
+
+    interface UserRegistrationData extends UserLoginData {
+        firstName: string;
+        lastName: string;
+        isLoggedin: boolean;
+        username?: string;
```
Comment (linablum) [inline server/src/@types/User.d.ts:15]: if username is optional, why have it? either none or obligatory I think

File: server/tsconfig.json (hunk 1 above)
Comment (podmar) [reply server/tsconfig.json]: removed

File: server/src/@types/User.d.ts (hunk 2 above)
Comment (podmar) [reply server/src/@types/User.d.ts:15]: for now I leave it the way it is, let's delete tomorrow

**Pull request merged**